libraries/grpc-sdk/src/classes/ConduitServiceModule.ts

```diff
@@ -1,9 +1,11 @@
 import ConduitGrpcSdk, {
   GrpcServer,
+  HealthCheckStatus,
 } from '..';
+import path from 'path';
 
 export abstract class ConduitServiceModule {
```

Comment:
Instead of simply basing further checks on overrides, functions can be exposed so that the module can do something like grpcSdk.updateHealth(Healthy);

Reply:
Fine by me, but we'd also be adding multi-service complexity to `ConduitServiceModule`, as modules exposing multiple services would need a way to update/check/watch these individually.

The class should be aware of the module name, supported service names, keep track of each service's state and handle cases where no service name is provided in a health check/watch request.
The last bit is not entirely obvious. Usually it'd make sense to respond <=non-serving as long as a any service is not serving, but what about if a module would still like to respond as `SERVING` when a non-critical service is offline etc?

I guess it could still just override the function at this point if anything other the normally expected behavior is desired.